### PR TITLE
Make sure other than user VMs can have multiple NICs in a network

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

During assignment of an IP from a secondary address range an extra NIC is created on a router. This must be allowed even though the router already has a NIC on the public network. The creation of extra NICs should only be prevented in User VMs on guest networks.

This PR...

Fixes: #5890

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

manual and throught the existing smoke tests.

### CODE REVIEW

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -4043,7 +4043,10 @@ public NicProfile addVmToNetwork(final VirtualMachine vm, final Network network,
     /**
      * duplicated in {@see UserVmManagerImpl} for a {@see UserVmVO}
      */
-    private void checkIfNetworkExistsForVM(VirtualMachine virtualMachine, Network network) {
+    private void checkIfNetworkExistsForUserVM(VirtualMachine virtualMachine, Network network) {
+        if (virtualMachine.getType() != VirtualMachine.Type.User) {
+            return; // others may have multiple nics in teh same network
```
Comment (weizhouapache) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:4048]: teh -> the 
^_^

**Comment by weizhouapache** [comment]
@DaanHoogland could you please update the title and description ?

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java (hunk 1 above)
Comment (DaanHoogland) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:4048]:
```suggestion
            return; // others may have multiple nics in the same network
```

**Comment by DaanHoogland** [comment]
@blueorangutan test keepEnv

**Comment by weizhouapache** [comment]
@DaanHoogland 
is this ready for testing ?

**Comment by DaanHoogland** [comment]
> @DaanHoogland is this ready for testing ?

yes @weizhouapache . I am pretty sure the HA failure has nothing to do with the code. I have tested locally.

**Comment by weizhouapache** [comment]
> > @DaanHoogland is this ready for testing ?
> 
> yes @weizhouapache . I am pretty sure the HA failure has nothing to do with the code. I have tested locally.

great. can you make it 'ready for review' ?

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by sureshanaparti** [comment]
Verified the fix with the steps mentioned in issue #5890.

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually checked

**Pull request merged**